Types for `.enu` should accept `null`

# Environment

Knex version: 1.0.5
Typescript: 4.6.3

Select applicable template from below.
@lorefnon.

# Bug

1. Explain what kind of behaviour you are getting and how you think it should do
The `.enum`/`.enu` functions of `TableBuilder` should accept `null` as a value input in the typings, as is the example in the documentation.

3. Error message
```
error TS2345: Argument of type 'null' is not assignable to parameter of type 'readonly Value[]'
```
4. Reduced test code, for example in https://npm.runkit.com/knex or if it needs real
   database connection to MySQL or PostgreSQL, then single file example which initializes
   needed data and demonstrates the problem.

```typescript
///@ts-check
/**param { import('knex').Knex} Knex */
export async function up(knex) {
   await knex.schema.createTable("mytable", (table) => {
     // typescript error occurs here:
      table.enu('column', null, { useNative: true, existingType: true, enumName: 'foo_type' })
   });
}
```

# Feature discussion / request

1. Explain what is your use case

2. Explain what kind of feature would support this

3. Give some API proposal, how the feature should work

--- a/test-tsd/tables.test-d.ts
+++ b/test-tsd/tables.test-d.ts
@@ -101,6 +101,12 @@ const main = async () => {
     table.integer('num').references('id').withKeyName('non_for7').deferrable('deferred').inTable('non_exist').onDelete('CASCADE');
     table.integer('num').references('id').inTable('non_exist').onDelete('CASCADE').withKeyName('non_for6').deferrable('deferred');
     table.integer('num').references('id').withKeyName('non_for7').onDelete('CASCADE').deferrable('deferred').inTable('non_exist');
+    
+    table.enu("myenum", null, {
+      enumName:"MyEnum",
+      useNative: true,
+      existingType: true,
+    });
 
     expectType<Knex.ReferencingColumnBuilder>(
       table.integer('num').references('id').withKeyName('non_for7').onDelete('CASCADE')

--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -2097,12 +2097,12 @@ export declare namespace Knex {
     binary(columnName: string, length?: number): ColumnBuilder;
     enum(
       columnName: string,
-      values: readonly Value[],
+      values: (readonly Value[]) | null,
       options?: EnumOptions
     ): ColumnBuilder;
     enu(
       columnName: string,
-      values: readonly Value[],
+      values: (readonly Value[]) | null,
       options?: EnumOptions
     ): ColumnBuilder;
     json(columnName: string): ColumnBuilder;

